parsec/core/gui/custom_dialogs.py

```diff
@@ -199,24 +224,39 @@ def on_finished(self):
         # is not set to None. Linux seems to clear them automatically over time.
         # Resetting the parent on MacOS causes a crash.
         if sys.platform != "darwin":
-            self.setParent(None)
+            # Mypy This is the correct way to remove a parent from a widget but PyQt miss that overload
+            self.setParent(None)  # type: ignore[call-overload]
+
+
+def exec_method(
```

Comment:
Maybe call it `generate_dialog_method` or something like that?

Reply:
done